exporter/prometheus/prometheus_test.go

```diff
@@ -183,3 +189,102 @@ func TestCollectNonRacy(t *testing.T) {
 		}
 	}()
 }
+
+type mCreator struct {
+	m   *stats.Int64Measure
+	err error
+}
+
+type mSlice []*stats.Int64Measure
+
+func (mc *mCreator) createAndAppend(measures *mSlice, name, desc, unit string) {
+	mc.m, mc.err = stats.Int64(name, desc, unit)
+	*measures = append(*measures, mc.m)
+}
+
+type vCreator struct {
+	v   *view.View
+	err error
+}
+
+func (vc *vCreator) createAndSubscribe(name, description string, keys []tag.Key, measure stats.Measure, agg view.Aggregation) {
+	vc.v, vc.err = view.New(name, description, keys, measure, agg)
+	if err := vc.v.Subscribe(); err != nil {
+		vc.err = err
+	}
+}
+
+func TestMetricsEndpointOutput(t *testing.T) {
+	exporter, err := newExporter(Options{})
+	if err != nil {
+		t.Fatalf("failed to create prometheus exporter: %v", err)
+	}
+	view.RegisterExporter(exporter)
+
+	names := []string{"foo", "bar", "baz"}
+
+	measures := make(mSlice, 0)
+	mc := &mCreator{}
+	for _, name := range names {
+		mc.createAndAppend(&measures, "tests/"+name, name, "")
+	}
+	if mc.err != nil {
+		t.Errorf("failed to create measures: %v", err)
+	}
+
+	vc := &vCreator{}
+	for _, m := range measures {
+		vc.createAndSubscribe(m.Name(), m.Description(), nil, m, view.CountAggregation{})
+	}
+	if vc.err != nil {
+		t.Fatalf("failed to create views: %v", err)
+	}
+	view.SetReportingPeriod(time.Millisecond)
+
+	for _, m := range measures {
+		stats.Record(context.Background(), m.M(1))
+	}
+
+	srv := httptest.NewServer(exporter)
+	defer srv.Close()
+
+	var i int
+	var output string
+	for output == "" {
```

Comment:
From the last review, we should have made this a plain `for`
as below we check `if output != "" { break }` but that's okay :)

Reply:
Ah, thanks for spotting that. Fixing now.